src/ledger/LedgerTxnAccountSQL.cpp

```diff
@@ -367,6 +370,348 @@ LedgerTxnRoot::Impl::deleteAccount(LedgerKey const& key)
+    st.define_and_bind();
+    {
+        auto timer = DB.getUpsertTimer("account");
+        st.execute(true);
+    }
+}
+
+template <typename T>
+std::string
+marshalToPGArrayItem(PGconn* conn, const T& item)
+{
+    std::ostringstream ss;
+    ss << item;
+    return ss.str();
+}
+
+template <>
+std::string
+marshalToPGArrayItem<std::string>(PGconn* conn, const std::string& item)
+{
+    static const std::string q("\"");
+    char buf[item.size() * 2 + 1];
+    int err = 0;
+    size_t len = PQescapeStringConn(conn, buf, item.c_str(), item.size(), &err);
+    if (err != 0)
+    {
+        throw std::runtime_error("Could not escape string in SQL");
+    }
+    return q + std::string(buf, len) + q;
+}
+
+template <typename T>
+std::string
+marshalToPGArray(PGconn* conn, const std::vector<T>& v,
+                 const std::vector<soci::indicator>* ind = 0)
+{
+    std::string result = "{";
+    for (size_t i = 0; i < v.size(); ++i)
+    {
+        if (i > 0)
+        {
+            result += ", ";
+        }
+        if (ind && (*ind)[i] == soci::i_null)
+        {
+            result += "NULL";
+        }
+        else
+        {
+            result += marshalToPGArrayItem(conn, v[i]);
+        }
+    }
+    result += "}";
+    return result;
+}
+
+static void
+postgresSpecificBulkUpsertAccounts(
+    Database& DB, std::vector<std::string> const& accountIDs,
+    std::vector<int64_t> const& balances, std::vector<int64_t> const& seqNums,
+    std::vector<uint32_t> const& subEntryNums,
+    std::vector<std::string> const& inflationDests,
+    std::vector<soci::indicator>& inflationDestInds,
+    std::vector<uint32_t> const& flags,
+    std::vector<std::string> const& homeDomains,
+    std::vector<std::string> const& thresholds,
+    std::vector<uint32_t> const& lastModifieds,
+    std::vector<int64_t> const& buyingLiabilities,
+    std::vector<int64_t> const& sellingLiabilities,
+    std::vector<soci::indicator>& liabilitiesInds)
+{
+
+    soci::session& session = DB.getSession();
+    auto pg =
+        dynamic_cast<soci::postgresql_session_backend*>(session.get_backend());
+    PGconn* conn = pg->conn_;
+
+    std::string accountIDstr = marshalToPGArray(conn, accountIDs);
+    std::string balancestr = marshalToPGArray(conn, balances);
+    std::string seqNumstr = marshalToPGArray(conn, seqNums);
+    std::string subEntryNumstr = marshalToPGArray(conn, subEntryNums);
+    std::string inflationDeststr =
+        marshalToPGArray(conn, inflationDests, &inflationDestInds);
+    std::string flagstr = marshalToPGArray(conn, flags);
+    std::string homeDomainstr = marshalToPGArray(conn, homeDomains);
+    std::string thresholdstr = marshalToPGArray(conn, thresholds);
+    std::string lastModifiedstr = marshalToPGArray(conn, lastModifieds);
+    std::string buyingLiabilitiestr =
+        marshalToPGArray(conn, buyingLiabilities, &liabilitiesInds);
+    std::string sellingLiabilitiestr =
```

Comment:
Nitpick: could we do a naming scheme like `strSellingLiabilities` instead? Prevents weird mangling of names. For example, hard to remember if it is `sellingLiabilitiestr` or `sellingLiabilitiesstr` or `sellingLiabilitiesStr`.